hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/security/client/ClientToAMTokenIdentifier.java

```diff
@@ -129,60 +124,28 @@ public String getClientName() {
             return proto;
         }
 
-        @Override
-        public Text getKind() {
-            return KIND_NAME;
-        }
-
-        @Override
-        public UserGroupInformation getUser() {
-            String clientName = getClientName();
-            if (clientName == null) {
-                return null;
-            }
-            return UserGroupInformation.createRemoteUser(clientName);
-        }
-
-        @Override
         public void write(DataOutput out) throws IOException {
             out.write(proto.toByteArray());
         }
-
-        @Override
-        public void readFields(DataInput in) throws IOException {
-            proto = YarnSecurityTokenProtos.ClientToAMTokenIdentifierProto.parseFrom((DataInputStream)in);
-        }
     }
 
-    public class ByteArray extends ClientToAMTokenIdentifier {
+    public class PojoImpl implements InternalTokenIdentifier {
         private ApplicationAttemptId applicationAttemptId;
         private Text clientName = new Text();
 
-        public ByteArray() {
-        }
-
-        public ByteArray(ApplicationAttemptId id, String client) {
-            this.applicationAttemptId = id;
-            this.clientName = new Text(client);
-        }
-
-        @Override
-        public void write(DataOutput out) throws IOException {
-            out.writeLong(this.applicationAttemptId.getApplicationId()
-                .getClusterTimestamp());
-            out.writeInt(this.applicationAttemptId.getApplicationId().getId());
-            out.writeInt(this.applicationAttemptId.getAttemptId());
-            this.clientName.write(out);
-        }
-
-        @Override
-        public void readFields(DataInput in) throws IOException {
+        PojoImpl(byte[] bytes) throws IOException {
+            DataInputStream in = new DataInputStream(new ByteArrayInputStream(bytes));
             this.applicationAttemptId =
                 ApplicationAttemptId.newInstance(
                     ApplicationId.newInstance(in.readLong(), in.readInt()), in.readInt());
             this.clientName.readFields(in);
         }
 
+        PojoImpl(ApplicationAttemptId id, String client) {
```

Comment:
is this used ?

Reply:
Yes in     public ClientToAMTokenIdentifier(ApplicationAttemptId id, String client) {